ttypdb/src/sidecar/count.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+var errProcStat = errors.New("broken process stat")
+
+// ttyCount returns the number of controlling terminals observed.
+// NOTE: This implementation is for Linux.
+func ttyCount() (int, error) {
+	dirs, err := os.ReadDir("/proc")
+	if err != nil {
+		return 0, err
+	}
+
+	ttys := map[string]bool{}
+	for _, d := range dirs {
+		err := func() error {
+			name := d.Name()
+			for _, ch := range name {
+				if ch < '0' || ch > '9' {
+					return nil
+				}
+			}
+
+			statBytes, err := os.ReadFile(filepath.Join("/proc", name, "stat"))
+			if err != nil {
+				return err
+			}
+
+			// The 6th (0-origin) field is controlling tty device number.
+			// If it is "0", the process is not controlled.
+			stat := string(statBytes)
+			idx := strings.LastIndexByte(stat, ')')
```

Comment:
It would be simpler if we could use `fmt.Sscanf()`.
Do we need to care about a command name with spaces?

Reply:
Yes, wee need to do so.
procps (`ps` command) do the same thing: https://gitlab.com/procps-ng/procps/-/blob/v4.0.4/library/readproc.c?ref_type=tags#L578-581